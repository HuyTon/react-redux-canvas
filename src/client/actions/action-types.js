// Canvas
export const GET_CANVAS_SUCCESS = 'GET_CANVAS_SUCCESS';

// Canvas Layout
export const LOAD_CANVAS_LAYOUT = 'LOAD_CANVAS_LAYOUT';

// Shapes
export const GET_SHAPES_SUCCESS = 'GET_SHAPES_SUCCESS';
export const DRAW_FULL_SHAPE = 'DRAW_FULL_SHAPE';
export const DRAW_SEMI_SHAPE = 'DRAW_SEMI_SHAPE';
export const REMOVE_SHAPE = 'REMOVE_SHAPE';
