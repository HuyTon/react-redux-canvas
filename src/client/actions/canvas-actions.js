import * as types from './action-types';

export const getCanvasSuccess = (canvas) => {
  return {
    type: types.GET_CANVAS_SUCCESS,
    canvas
  };
}