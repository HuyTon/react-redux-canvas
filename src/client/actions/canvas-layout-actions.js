import * as types from './action-types';

export const loadCanvasLayout = (title) => {
    return {
        type: types.LOAD_CANVAS_LAYOUT,
        title
    };
}
