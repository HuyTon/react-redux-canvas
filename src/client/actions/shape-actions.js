import * as types from './action-types';

export const getShapesSuccess = (shapes) => {
  return {
    type: types.GET_SHAPES_SUCCESS,
    shapes
  };
}

export const drawFullShape = (shapeId) => {
  return {
    type: types.DRAW_FULL_SHAPE,
    shapeId
  };
}

export const drawSemiShape = (shapeId) => {
  return {
    type: types.DRAW_SEMI_SHAPE,
    shapeId
  };
}

export const removeShape = (shapeId) => {
  return {
    type: types.REMOVE_SHAPE,
    shapeId
  };
}