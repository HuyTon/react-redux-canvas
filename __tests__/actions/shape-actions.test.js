import * as shapeActions from '../../src/client/actions/shape-actions';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore({});

describe('shape_actions', () => {
    beforeEach(() => {
        store.clearActions();
    });
    describe('getShapesSuccess', () => {
        test('Dispatches the correct action getShapesSuccess', () => {
            store.dispatch(shapeActions.getShapesSuccess('shapes'));
            expect(store.getActions()).toMatchSnapshot();
        });
    });
    describe('drawFullShape', () => {
        test('Dispatches the correct action drawFullShape', () => {
            store.dispatch(shapeActions.drawFullShape('1'));
            expect(store.getActions()).toMatchSnapshot();
        });
    });
    describe('drawSemiShape', () => {
        test('Dispatches the correct action drawSemiShape', () => {
            store.dispatch(shapeActions.drawSemiShape('1'));
            expect(store.getActions()).toMatchSnapshot();
        });
    });
    describe('removeShape', () => {
        test('Dispatches the correct action removeShape', () => {
            store.dispatch(shapeActions.removeShape('1'));
            expect(store.getActions()).toMatchSnapshot();
        });
    });
})