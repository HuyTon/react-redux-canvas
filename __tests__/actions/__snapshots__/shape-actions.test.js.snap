// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shape_actions drawFullShape Dispatches the correct action drawFullShape 1`] = `
Array [
  Object {
    "shapeId": "1",
    "type": "DRAW_FULL_SHAPE",
  },
]
`;

exports[`shape_actions drawSemiShape Dispatches the correct action drawSemiShape 1`] = `
Array [
  Object {
    "shapeId": "1",
    "type": "DRAW_SEMI_SHAPE",
  },
]
`;

exports[`shape_actions getShapesSuccess Dispatches the correct action getShapesSuccess 1`] = `
Array [
  Object {
    "shapes": "shapes",
    "type": "GET_SHAPES_SUCCESS",
  },
]
`;

exports[`shape_actions removeShape Dispatches the correct action removeShape 1`] = `
Array [
  Object {
    "shapeId": "1",
    "type": "REMOVE_SHAPE",
  },
]
`;
